modernize a little autotools

sergiomb2 opened the pull request:
Since aMule are about to have a new version, here is my contribution , autopoint  may be replace by a newer intltool, is a more elegant solution .

gonosztopi commented:
The empty files your patch adds are already there in the docs subdir - and should stay there. That's why we have the `--foreign` switch on automake.

sergiomb2 commented:
OK back to
```
aclocal -I m4
autoheader
autoconf
automake --foreign -a -c -f
```

autoreconf doesn't accept foreign option ?!?

sergiomb2 commented:
I had use intltool-update to update po files , use relative paths , but still not fix the issue and using IT_PROG_INTLTOOL we don't have translation and back to AM_GNU_GETTEXT we have  translation, anyway we have some improves ...

gonosztopi commented:
I still have complaints... :-)

Change of all source references in translation files - should be done in a separate commit.
Changing "don't" to "dont" - not accepted.
Now where is the 'intl' subdirectory, needed on systems not having GNU gettext?

sergiomb2 commented:
> Changing "don't" to "dont" - not accepted.

The single quotes doesn't break yours syntax highlighting ? can I use "do not" ?

gonosztopi commented:
> The single quotes doesn't break yours syntax highlighting ? can I use "do not" ? 
No, not mine :-) Anyway, I'd rather have a broken syntax highlighting than syntactically wrong messages to the user.

sergiomb2 commented:
can I use "do not" ?

sergiomb2 commented:
```
 AM_GNU_GETTEXT_VERSION(0.11.5)
+AM_GNU_GETTEXT([external])
+#IT_PROG_INTLTOOL
+#GETTEXT_PACKAGE="amule"
+#AC_SUBST(GETTEXT_PACKAGE)
+#AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE",[Translation domain used])
```
if I use IT_PROG_INTLTOOL  and next commands instead AM_GNU_GETTEXT (that is what I want to do ) , it compiles and builds .
But if I use GETTEXT_PACKAGE="AC_PACKAGE_NAME" , find_lang requires use aMule with capital M 
if I use GETTEXT_PACKAGE="amule" find_lang amule doesn't complains , but in both cases , applications translations doesn't work , have any idea what might be ? 
Thanks

timofonic commented:
Any news?

sergiomb2 commented:
> Any news?

I think is finish , you got the final PR , after some rounds we just run "autoreconf --install" in autogen.sh , autoreconf will run autopoint,  and others autos and will not produce any warning. 

[TIL : that is intltool that is deprecated not gettext](https://wiki.gnome.org/MigratingFromIntltoolToGettext) , now I have revert intltoollize in others projects 😃

gonosztopi reviewed (changes requested):
`subdir-objects` option to `AM_INIT_AUTOMAKE` is potentially dangerous and should not be used. 

Although it has its own benefits, aMule source code is organized such that it requires the same file being compiled multiple times with potentially different compiler flags. Using `subdir-objects` would compile the affected files only for the first time with the compiler flags of the first target, and assuming it's valid for all other targets too. Before using `subdir-objects` there have to be some source reorganization and build rule review.

sergiomb2 commented:
> subdir-objects option to AM_INIT_AUTOMAKE is potentially dangerous and should not be used.

I'm not aware of any problem with  subdir-objects [1] , add it because silent a lot of warnings like: 

```
src/Makefile.am:72: warning: source file 'kademlia/utils/UInt128.cpp' is in a subdirectory,
src/Makefile.am:72: but option 'subdir-objects' is disabled
```

so I just enable subdir-objects ! , fixed now to merge all other modifications 

[1] https://www.gnu.org/software/automake/manual/html_node/List-of-Automake-options.html#List-of-Automake-options

sergiomb2 commented:
Review done with: 
https://github.com/amule-project/amule/pull/43/commits/7ce79a36aed056fe25282de2dbfda4742cea87a8

gonosztopi reviewed (commented):
There is still one thing to mention.

Although we use wxWidgets' implementation of the gettext() routinges almost everywhere, the PHP interpreter built into the webserver still uses the library gettext() function. Though I think nobody uses that, to compile the webserver we need to ensure there's a working library gettext() function on every system we support. That's why the intl subdirectory is there.

Of course the solution would be to make the PHP gettext functions use wxWidgets calls. (And properly add translation support for templates.) But until that happens, this patch cannot be merged.

timofonic commented:
@sergiomb2 Any news about this? Many months have passed...

sergiomb2 commented:
Nothing change much , I found in the middle of the process , I learned that is intltool that is deprecated not gettext , so this Pull request have much less things to update .

> subdir-objects option to AM_INIT_AUTOMAKE is potentially dangerous and should not be used.

so we still got a lot of warnings 

> There is still one thing to mention.
> 
> Although we use wxWidgets' implementation of the gettext() routinges almost everywhere, the PHP interpreter built into the webserver still uses the library gettext() function. Though I think nobody uses that, to compile the webserver we need to ensure there's a working library gettext() function on every system we support. That's why the intl subdirectory is there.
> 
> Of course the solution would be to make the PHP gettext functions use wxWidgets calls. (And properly add translation support for templates.) But until that happens, this patch cannot be merged.

sergiomb2 commented:
OK I finished this review , gettext works perfectly , only thing I'm not sure is if you shouldn't move License 
ChangeLog , AUTHORS etc to the root directory , seems that is a rule of autotools ... 
Best regards,

sergiomb2 commented:
> only thing I'm not sure is if you shouldn't move License
ChangeLog , AUTHORS etc to the root directory , seems that is a rule of autotools ...

I just found in other project that we may use symbol links ...

sergiomb2 commented:
even more simplified , 3 commit only wit delete file one commit [1] with modified code 

[1]
https://github.com/amule-project/amule/pull/43/commits/8f5d1453c67b5f9854975336189a1b3949ac469b

sergiomb2 commented:
ping only https://github.com/amule-project/amule/pull/43/commits/8f5d1453c67b5f9854975336189a1b3949ac469b have code , the other 3 commit is just deleting files 
and renamed:    m4/nls.m4 -> m4/mulenfs.m4

sergiomb2 committed:
Add AC_CONFIG_AUX_DIR([config]) to configure.ac

sergiomb2 committed:
Add subdir-objects to AM_INIT_AUTOMAKE

sergiomb2 commented:
rebased with upstream/master and fixed the divergences

sergiomb2 committed:
use actions/checkout@v4 and install autopoint and autoconf

sergiomb2 committed:
Fix some deprecated autotools warnings

autoupdate

AC_FUNC_SETVBUF_REVERSED the macro is no longer needed

AC_HELP_STRING is now deprecated but can be replaced with AS_HELP_STRING

sergiomb2 committed:
Simplify autogen.sh using autoreconf only (v 2024)

Copy ./AUTHORS ./ChangeLog ./COPYING from docs to root because they are
required required files by autotools

Touch NEWS

Rename m4/nls.m4 to m4/mulenfs.m4 to avoid collisions with real nls.m4

Delete many generated files in m4/, po/ and docs/ABOUT-NLS and also add
these files to .gitignore

Delete docs/license.txt because is equal to COPYING

mrjimenez commented:
Hi @sergiomb2 , do you still want this patch to come in? Resolve the conflicts and I can approve it.

Regards,

mrjimenez commented:
Closing since we no longer use autotools.
